### ISSUE TITLE
Refactor checkIfAttachedToNode to return different error type based on error

### ISSUE THREAD

**Issue by gnufied** [kubernetes/kubernetes#59601]
Currently, this function just returns same `fmt.Errorf` no matter what error is. I think, it is a good idea to refactor this function to return different types of errors so as caller of this function can use the error to differentiate between errors and take action.

cc @tsmetana 
/sig aws
/sig storage

**Comment by tsmetana** [comment]
Yes, this needs to be fixed. We actually lose the awsError deeper in the stack, so I will probably have to change describeVolume wrapper and all its callers.
